polaris-react/src/components/Button/Button.scss

```diff
@@ -1,1043 +1,352 @@
-
-  @media #{$p-breakpoints-md-up} {
-    padding-left: var(--p-space-100);
-    padding-right: var(--p-space-100);
-  }
-
-  &.sizeLarge {
-    padding-left: var(--p-space-200);
-    padding-right: var(--p-space-200);
-
-    @media #{$p-breakpoints-md-up} {
-      padding-left: var(--p-space-150);
-      padding-right: var(--p-space-150);
-    }
-  }
-
-  &.sizeSlim {
-    padding-left: var(--p-space-100);
-    padding-right: var(--p-space-100);
-  }
-
-  .Icon:first-child {
-    margin-left: 0;
-  }
-
-  .Icon:last-child {
-    margin-right: calc(-1 * var(--p-space-100));
-  }
-
-  .Icon:only-child {
-    margin-right: 0;
-  }
+  --pc-button-padding-inline: var(--p-space-100);
 }
 
-.sizeMicro {
-  border-radius: var(--p-border-radius-200);
-  padding: var(--p-space-100) var(--p-space-200);
-  // stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom property
-  min-height: var(--pc-button-micro-min-height);
-
-  .Content {
-    font-size: var(--p-font-size-325);
-    font-weight: var(--p-font-weight-medium);
-    line-height: var(--p-font-line-height-500);
-  }
-
-  @media #{$p-breakpoints-md-up} {
-    padding: var(--p-space-100) var(--p-space-200);
-    // stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom property
-    min-height: var(--pc-button-micro-min-height);
-
-    .Content {
-      font-size: var(--p-font-size-300);
-      line-height: var(--p-font-line-height-400);
-    }
-  }
+.iconOnly:is(.sizeLarge) {
+  --pc-button-padding-inline: var(--p-space-150);
 }
 
-.sizeSlim {
-  // stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom property
-  min-height: var(--pc-button-slim-min-height);
-  padding: var(--p-space-150) var(--p-space-300);
-
-  &.iconOnly {
-    padding-left: var(--p-space-150);
-    padding-right: var(--p-space-150);
-
-    @media #{$p-breakpoints-md-up} {
-      padding-left: var(--p-space-100);
-      padding-right: var(--p-space-100);
-    }
-  }
+.iconOnly:is(.sizeMicro) {
+  --pc-button-padding-inline: var(--p-space-050);
 }
 
-.sizeLarge {
-  /* stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom height */
-  min-height: var(--pc-button-large-min-height);
-  /* stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- custom height */
-  min-width: var(--pc-button-large-min-height);
-  padding-block: var(--p-space-200);
-  padding-inline: var(--p-space-300);
-
-  .Content {
-    font-size: var(--p-font-size-325);
-    font-weight: var(--p-font-weight-medium);
-    line-height: var(--p-font-line-height-500);
-    letter-spacing: initial;
-
-    @media #{$p-breakpoints-md-up} {
-      line-height: var(--p-font-line-height-400);
-    }
-  }
+.iconOnly:is(.variantTertiary) {
+  --pc-button-icon-fill: var(--p-color-icon-secondary);
+  --pc-button-icon-fill_hover: var(--p-color-icon);
```

Comment:
Should the pressed/active and hover states of tertiary icon only buttons use the `icon-secondary-x` tokens?

```suggestion
  --pc-button-icon-fill_hover: var(--p-color-icon-secondary-hover);
  --pc-button-icon-fill_active: var(--p-color-icon-secondary-active);
```

|| `--p-color-icon-x` | `--p-color-icon-secondary-x` |
|--------|--------|--------|
|(static for comparison)|![Screenshot 2023-12-13 at 12 30 12 PM](https://github.com/Shopify/polaris/assets/18447883/e11b06fc-af30-43fc-8316-f7455e022a46) | ![Screenshot 2023-12-13 at 12 30 12 PM](https://github.com/Shopify/polaris/assets/18447883/e11b06fc-af30-43fc-8316-f7455e022a46)|
|hover| ![Screenshot 2023-12-13 at 12 24 36 PM](https://github.com/Shopify/polaris/assets/18447883/bfe347be-0743-49f2-9d2f-3c3a72a9ec46) | ![Screenshot 2023-12-13 at 12 25 16 PM](https://github.com/Shopify/polaris/assets/18447883/525544da-dc99-49e2-acb2-6a6ba114efcf) |
|pressed/active| ![Screenshot 2023-12-13 at 12 20 16 PM](https://github.com/Shopify/polaris/assets/18447883/21b81da8-f89d-45f8-90a2-8c25bbd63494) | ![Screenshot 2023-12-13 at 12 21 06 PM](https://github.com/Shopify/polaris/assets/18447883/093ae019-517b-4a5d-8507-e19fe2b10595)|

Reply:
Yes. This makes more sense.